include/boost/program_options/parsers.hpp

```diff
@@ -120,7 +120,10 @@ namespace boost { namespace program_options {
         basic_command_line_parser(const std::vector<
                                   std::basic_string<charT> >& args);
         /** Creates a command line parser for the specified arguments
-            list. The parameters should be the same as passed to 'main'.
+            list. The parameters should be the same as passed to 'main', meaning:
+            @param argc Must be non-zero i.e. >= 0
+            @param argv argv[0] up to argv[argc-1] must point to null terminated strings and
+            argv[argc] must be 0 e.g. nullptr
```

Comment:
why was this added here? It appears as though this restriction is never relied on here (in fact neither `argc` or `argv` are in sight). Perhaps this part of POSIX semantics should be documented elsewhere?

Reply:
I did not refer to posix but the [C++20 draft N 4860](https://isocpp.org/files/papers/N4860.pdf) and https://eel.is/c++draft/basic.start.main#2.2 in the bug report: 
To quote 6.9.3.1 main function [basic.start.main] (2.2) from the draft

> If argc is nonzero these arguments shall be supplied in argv[0] through argv[argc-1] as pointers to the initial characters of null-terminated multibyte strings (ntmbss) (16.4.2.2.5.2) and argv[0] shall be the pointer to the initial character of a ntmbs that represents the name used to invoke the program or "". The value of argc shall be non-negative. The value of argv[argc] shall be 0.

or in [the newer one](https://eel.is/c++draft/basic.start.main#2.2)

So this comment shall make a regression of that bug less likely and state an API contract for function users and boost code contributors, what to expect.

Reply:
Ah. My confusion was due to github UI putting the comment "together" with the preceding constructor overload. I was struggling to see how it related there :) Disregard